wpcom-legacy-redirector.php

```diff
@@ -70,9 +70,10 @@ static function maybe_do_redirect() {
  	 *
  	 * @param string $from_url URL or path that should be redirected; should have leading slash if path.
  	 * @param int|string $redirect_to The post ID or URL to redirect to.
- 	 * @return bool|WP_Error Error if invalid redirect URL specified or if the URI already has a rule; false if not is_admin, true otherwise.
+ 	 * @param bool $mode add|update if "update", we will update existing redirects for from_url, otherwise we raise an error on an existing redirect
+ 	 * @return bool|WP_Error Error if invalid redirect URL specified or if the URI already has a rule and we aren't updating or if the update failed; false if something is very wrong (not admin, not CLI), true otherwise.
  	 */
-	static function insert_legacy_redirect( $from_url, $redirect_to ) {
+	static function insert_legacy_redirect( $from_url, $redirect_to, $mode='add' ) {
```

Comment:
I think the param might be better as a bool like `$update_if_exists` (I'm generally not a fan of booleans as parameters but I think it's okay here.)